source/Outfit.cpp

```diff
@@ -244,6 +263,29 @@ void Outfit::Load(const DataNode &node)
 
 
 
+void Outfit::FinishLoading()
+{
+	const Outfit *parent = GameData::Outfits().Get(parentName);
```

Comment:
This outfit might not have a parent, in which case you're grabbing a null object and referencing it if this outfit for some reason is missing any of the following attributes. This can actually happen in vanilla, as H2H weapons have no mass, in which case you're referencing a non-existent parent outfit for its mass. I imagine that's not good.

You should be checking if a parent exists before referencing it. Again, look to how ship does it.
https://github.com/endless-sky/endless-sky/blob/c9a79a3080fc7a52d1123385f5fdafc451fc73f3/source/Ship.cpp#L476-L486

Reply:
I would prefer a `Get` is called at some point to facilitate reference checking. If `Get` is used, it will never return a null pointer, but it may return a "garbage" pointer to an Outfit that has not been defined.

(Even if `Find` is used and the returned pointer is not null, the outfit may not yet have been loaded, so a defined-ness check is still required.)